hack/configure-kcp.sh

```diff
@@ -44,7 +44,14 @@ configure_compute_workspace() {
   SYNC_TARGET=appstudio-internal
   if [[ -z "$(kubectl get synctargets.workload.kcp.dev ${SYNC_TARGET} --kubeconfig ${KCP_KUBECONFIG} 2>/dev/null)" ]]; then
     echo "Creating SyncTarget..."
-    KUBECONFIG=${KCP_KUBECONFIG} kubectl kcp workload sync ${SYNC_TARGET} --syncer-image ghcr.io/kcp-dev/kcp/syncer:v0.9.0 --resources=services,routes.route.openshift.io -o /tmp/${SYNC_TARGET}-syncer.yaml
+    # CAUTION: '--resource=statefulsets.apps' is temporarily added as a synchronized resource to the syncer, until the GitOps 
+    # Service team moves Argo CD out of KCP. The KCP team does not support StatefulSets as of this writing (Oct 2022).
+    # 
+    # Until KCP adds support, StatefulSets should not be used by components except in the short term and in limited contexts,
+    # and with full understanding of the consequences. - @jgwest
+
+    KUBECONFIG=${KCP_KUBECONFIG} kubectl kcp workload sync ${SYNC_TARGET} --syncer-image ghcr.io/kcp-dev/kcp/syncer:v0.9.0 --resources=services,routes.route.openshift.io,statefulsets.apps -o /tmp/${SYNC_TARGET}-syncer.yaml
```

Comment:
I thought that StatefulSets are not supported by kcp. :man_shrugging:

Reply:
This is the story for moving Argo CD off KCP: https://issues.redhat.com/browse/GITOPSRVCE-237

Once complete,  we will be able to remove statefulsets from the syncer.

Reply:
hmm... I thought that the main goal was to run the ArgoCD instance(s) in a kcp workspace, not directly in an OSD cluster. So I'm surprised that you mentioned that you want to move it off kcp...
cc @alexeykazakov 

As for StatefulSet - when you did the experiment with running ArgoCD in a kcp workspace, you didn't mention anything related to unsupported StatefulSets, so this is new to me as well. 
Have you talked to kcp folks regarding this particular use-case of StatefulSet? If it's OK to go with that?
I'm slightly afraid of using something that is not supported by kcp, because we know that there can be some unexpected behavior even with the supported types - not sure what will happen with our syncer/workspace/wokload-cluster when using the unsupported ones.

Reply:
Discussed here: https://coreos.slack.com/archives/C02CTEB3MMF/p1662580655163479

Reply:
> I'm slightly afraid of using something that is not supported by kcp, 

We are as well, GITOPSRCE-237 will allow us to avoid this, by running Argo CD on workload cluster, rather than via KCP.

Reply:
@MatousJobanek, yes, while stateful sets work well enough for ArgoCD atm they may break any moment. However because the work has already been done it was decided to proceed and add the already tested and working manifests to the infra-deployment to deploy ArgoCD via KCP. That will allow us to unblock development which requires a running ArgoCD. Meanwhile the team will work on https://issues.redhat.com/browse/GITOPSRVCE-237.